src/reactapp/src/components/paypalExpress/hooks/usePaypalExpress.js

```diff
@@ -0,0 +1,138 @@
+import { useCallback, useContext, useEffect, useState } from 'react';
+import AppContext from '../../../context/App/AppContext';
+import createCustomerToken from '../../../api/paypalExpress/createCustomerToken';
+import { _objToArray } from '../../../utils';
+import { __ } from '../../../i18n';
+import setPaymentMethodPaypalExpress from '../../../api/paypalExpress/setPaymentMethod';
+import placeOrder from '../../../api/cart/placeOrder';
+import useShippingMethodCartContext from '../../shippingMethod/hooks/useShippingMethodCartContext';
+import LocalStorage from '../../../utils/localStorage';
+import { config } from '../../../config';
+import usePaymentMethodCartContext from '../../paymentMethod/hooks/usePaymentMethodCartContext';
+import useBillingAddressCartContext from '../../billingAddress/hooks/useBillingAddressCartContext';
+
+export default function usePaypalExpress() {
+  const [, appActions] = useContext(AppContext);
+  const { setErrorMessage, setPageLoader } = appActions;
+  const { selectedMethod } = useShippingMethodCartContext();
+  const { cartBillingAddress } = useBillingAddressCartContext();
+  const query = window.location.href.split('?')[1];
+  const [processPaymentEnable, setProcessPaymentEnable] = useState(false);
+  const { selectedPaymentMethod, methodList } = usePaymentMethodCartContext();
+  const cartId = LocalStorage.getCartId();
+
+  /*
+   Check if is possible to proceed on placing the order.
+   */
+  useEffect(() => {
+    if (
+      query &&
+      selectedMethod?.methodCode &&
+      (selectedPaymentMethod?.code === '' ||
```

Comment:
may be we can combine this condition like `['', 'paypal_express'].includes(selectedPaymentMethodCode)`. Remember again `paypal_express` hardcode here. we need to pass `method.code` to the hook and use it.

Reply:
Great, done